cmake/spl.cmake

```diff
@@ -22,20 +22,28 @@ if(EXISTS ${AUTOCONF_CMAKE})
 endif()
 
 if(BUILD_KIT STREQUAL prod)
-    # set default link target output name
-    if (LINK_FILE_BASENAME)
-        message("LINK_FILE is set to '${LINK_FILE_BASENAME}' and will be used.")
-    else()
-        set(LINK_FILE_BASENAME main)
-    endif()
+    # set default link target output name and extension
+    if (LINKER_OUTPUT_FILE)
+        get_filename_component(LINK_FILE_BASENAME ${LINKER_OUTPUT_FILE} NAME_WE)
+        message("LINK_FILE_BASENAME is set to '${LINK_FILE_BASENAME}' and will be used.")
 
-    # set default link target output name
-    if (LINK_FILE_EXTENSION)
+        get_filename_component(LINK_FILE_EXTENSION ${LINKER_OUTPUT_FILE} EXT)
         message("LINK_FILE_EXTENSION is set to '${LINK_FILE_EXTENSION}' and will be used.")
     else()
+        set(LINK_FILE_BASENAME main)
         set(LINK_FILE_EXTENSION .exe)
     endif()
 
+    #combine basename and byproduct extension
+    set(LINKER_BYPRODUCTS "")
+    string(REPLACE "," ";" LINKER_BYPRODUCTS_EXTENSIONS "${LINKER_BYPRODUCTS_EXTENSIONS}")
+    foreach(extension ${LINKER_BYPRODUCTS_EXTENSIONS})
+        string(APPEND LINKER_BYPRODUCTS "${LINK_FILE_BASENAME}.${extension};")
+    endforeach()
+    string(REGEX REPLACE ";$" "" LINKER_BYPRODUCTS "${LINKER_BYPRODUCTS}")  # Remove trailing semicolon

```

Comment:
Although ChatGPT proposed this it is not very nice nor efficient. I propose to replace it with this:

```suggestion
    #combine basename and byproduct extension
    string(REPLACE "," ";" LINKER_BYPRODUCTS "${LINKER_BYPRODUCTS_EXTENSIONS}")
    list(TRANSFORM LINKER_BYPRODUCTS PREPEND ${LINK_FILE_BASENAME}.)
```